### ISSUE TITLE
Screen Reader: Many Pages: Date picker controls are announced without a button role

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77561]
## Action Performed:
Precondition: User is logged in

1. Using VoiceOver on iOS, open the app
2. Swipe right to focus the Account tab and double-tap to activate it
3. Swipe right to focus the Profile button and double-tap to activate it
4. Swipe right to focus the Date of birth menu item and double-tap to activate it
5. When the Date of birth picker opens, navigate through the interactive elements of the calendar and note how they are announced

Other Occurrences:
1. Create Expense > Confirm details > Date: same issue
2. Reports > Filters > Date > On/After/Before: same issue

## Expected Result:
All interactive elements in the date picker should be announced with a button role.

## Actual Result:
The year selector, previous/next month arrows, and individual calendar dates are announced without a role.
Screen readers announce them only as text ("12", "Current year", "Next"), making it unclear that they are actionable.

## Area issue was found in:
Profile Subsections

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Apply the accessibilityRole="button" prop to every interactive date picker control to ensure they are announced with the button role.
https://appt.org/en/docs/react-native/samples/accessibility-role
https://reactnative.dev/docs/accessibility#accessibilityrole

## Environment:
Apple iPhone 14 Pro - iOS 18.7.1, Wi-Fi - Native (No mobile browser)

## Screenshots/Videos

<img width="1179" height="2438" alt="Image" src="https://github.com/user-attachments/assets/132b78bd-e306-4b39-9f25-22c0d3767479" />

<img width="1179" height="2433" alt="Image" src="https://github.com/user-attachments/assets/aebc8ced-7b7a-4c76-8981-e3d6b886f8c3" />

![Image](https://github.com/user-attachments/assets/2fc09012-6f94-4303-9837-effc2cd35fd5)

https://github.com/user-attachments/assets/6138a998-96e5-419c-89eb-75d1322e6bc5

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7005787
**Issue reported by:** NA
**Slack conversation:** NA
